### DISCUSSION TITLE
onLeaveScene callback/event

### DISCUSSION THREAD

**Discussion by dbarnett** [replit/kaboom#608] [Ideas]
Would it be okay to have a hook that gets called on leaving a scene? There are some kinds of cleanup like stopping music or conditionally destroying objects with `stay()` that get messy trying to do ad-hoc cleanup before any code leaves the scene.

I could imagine this offered either via another `SceneOpts` arg to the `scene()` function or as an `onLeaveScene(id, action)` event helper.

**Comment by lajbel** [comment]
Very good idea! Up Voted

**Comment by slmjkdbtl** [comment]
Added in f80392b
